src/matrixeventprocessor.ts

```diff
@@ -92,6 +98,33 @@ export class MatrixEventProcessor {
         return body;
     }
 
+    public ReplaceDiscordEmoji(content: string, guild: Discord.Guild): string {
+        let results = DISCORD_EMOJI_REGEX.exec(content);
+        while (results !== null) {
+            const emojiName = results[1];
+            const emojiNameWithColons = results[0];
+
+            // Check if this emoji exists in the guild
+            if(guild.emojis[emojiName] !== null) {
+                // Replace :a: with <:a:123ID123>
+                const emojiID = this.FindDiscordEmojiIDWithName(emojiName, guild.emojis);
```

Comment:
I'd have just used ``guild.emojis.find((emojiCollection) => { ... });`` but I guess either is fine.

Reply:
You and your JS knowledge. I'll try that, thanks :)